src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs

```diff
@@ -253,6 +255,21 @@ protected static bool IsInvalidArgument(IReadOnlyList<object> list)
             {
                 await task(message);
             }
+            catch (TimeoutException ex)
+            {
+                string connectionId = null;
+                if (message is IConnectionAckableMessage msg)
+                {
+                    connectionId = msg.ConnectionId;
+                }
+
+                // Regard the case connection already disconnected when send group message got timeout as success
+                if (connectionId == null || _clientConnectionManager.ClientConnections.TryGetValue(connectionId, out var connection))
```

Comment:
this logic can be moved into L263

Reply:
Considering this is a common method and would like to cover most situation. Do you suggest to do like this?
```
string connectionId = null;
// connection still exists and meet time-out.
if (message is IConnectionAckableMessage msg)
{
    if (_clientConnectionManager.ClientConnections.TryGetValue(connectionId, out var connection))
    {
        Log.FailedToSendMessage(Logger, message, ex);
        throw;
    }
}

// other ackable messages meet time-out
if (connectionId == null)
{
    Log.FailedToSendMessage(Logger, message, ex);
    throw;
}
```